### DISCUSSION TITLE
Unable to Publish New Meetings

### DISCUSSION THREAD

**Discussion by LACSupport** [code4recovery/12-step-meeting-list#480] [Q&A] [answered]
I have 61 published meetings, however just recently (within a week) I have not been able to Publish new meetings. When I try to create a new meeting, I enter in all the details as I normally would, however the "Publish" button is grayed out, and it automatically saves it as a draft. I am also unable to update the automatically saved draft. I have tried backing out into the list of meetings to edit the status manually to "Publish", however this does not resolve the issue. When I try to preview the meeting, the status shows as "Temporarily Inactive".

 - OS: macOS Big Sur 11.4
 - Chrome Browser
 - Version 92.0.4515.107

**Comment by tech2serve** [comment, marked as answer]
Most likely this is an issue with geocoding. Something has changed with your configuration perhaps?

In your browser, open up the developer's console and watch for errors when you enter the address for the meeting. Note any errors and bring them back here if you aren't able to figure it out yourself.

On the plugin's Import & Settings page, note what is entered for the Geocoding Method:

1. **Legacy**--Should be okay, but if you're getting an error, we need to figure out what is causing it. One possibility is a conflict with another plugin.
2. **Google**--This should only show as an option if you've entered a Google API key above. This is really an advanced user option, and should generally be avoided unless you really know what you're doing. You have to configure the API key through the Google's developer console specifically for geocoding in addition to mapping. If you choose to use this, you'll need to login into your Google Developer's console to troubleshoot geocoding issues. 
3. **Beta API Gateway**--This is the preferred option because it helps us test out our system in development and gives us a better ability to troubleshoot behind the scenes. 

For most users, the simplest configuration is Mapbox (access token) for mapping and either **Legacy** or Code for Recovery's **API Gateway** for geocoding. You can still configure the system to use Google for mapping if you wish, but it is still easier to use options 1 or 3 for geocoding. 

Make sure you're configured with one of those options and the problem should clear up. If it doesn't, come back with the displayed errors.

**Reply by LACSupport** [reply]
Hello, and thank you for your response; here are the errors that show up:

https://lynnwoodalanoclub.org/?post_type=tsml_meeting&p=8471
![Screen Shot 2021-07-30 at 7 13 12 PM](https://user-images.githubusercontent.com/88159477/127725662-49aa278f-7b42-40c4-8dc4-5817e762d445.png)

To answer your second question, the Geocoding Method is set to Legacy.

Thank you again for your assistance with this!

**Reply by LACSupport** [reply]
This is the error that popped up when I entered the address within the Meetings plugin in WordPress:

![Screen Shot 2021-07-30 at 7 23 36 PM](https://user-images.githubusercontent.com/88159477/127725922-212e9053-d012-497a-8548-42c2ced62df0.png)

**Reply by LACSupport** [reply]
I updated from Legacy to Beta API Gateway, and it finally allowed me to publish the meeting, however there's still an error with the map I think. It doesn't show an actual map, just a red location icon.

**Reply by tech2serve** [reply]
Yeah. It looks like something is wrong with your API key with Google. If you want to continue to use Google, you'll need to log into that account and troubleshoot from there.

Or, simpler perhaps, just create a Mapbox account and generate and access token. Use that instead of Google. 

Your choice though. 

Once you get the mapping solved, I suggest looking in the console again to make sure errors are all cleared up.

Good luck. Glad the geocoding issue is resolved.

**Reply by LACSupport** [reply]
Thank you very much, using MapBox worked! I used the default public access token, added the API code to the meetings settings, deactivated Google Maps plugin and all works well now! Thank you very much.

**Reply by AZ-AA-Ron** [reply]
Hello! I am also having this issue where after publishing several meetings the "publish" button is now grayed out and I am unable to make changes/ add new meetings. 
I am using OS 12.6.2 with Safari 12.6.2. 
We use a Mapbox Access Token. Here's a screenshot of the console when searching for an address with a known street name:
![8F35882C-2694-4688-A1B9-E6CF96F50665](https://github.com/code4recovery/12-step-meeting-list/assets/137969050/5c1238e8-49b3-4d59-a0ff-dfe66c6577d6)
![8F35882C-2694-4688-A1B9-E6CF96F50665](https://github.com/code4recovery/12-step-meeting-list/assets/137969050/2bb72d3f-d0d8-40c5-a862-c1cf5d886e31)

**Reply by joshreisner** [reply]
hi @AZ-AA-Ron in the screenshot it looks like you're searching for meetings matching "soliere" - i see three meetings on your frontend https://flagstaffaa.org/meetings/?tsml-day=any&tsml-query=soliere - i believe the search on the frontend may be more thorough (searches more fields)

if you're having problems publishing new meetings - could you share a screenshot of that?

**Comment by AZ-AA-Ron** [comment]
Hi there! Thanks for getting back to me. On the "Soliere" search I was
trying to follow the suggestions given to the previous poster with the same
issue.

Anyways--here's a few screenshots:
Let's say I want to edit a draft for this meeting--this is how it first
appears (notice "publish" button is still blue):
[image: image.png]

I want to edit the status of in-person and make it online so I change it:
[image: image.png][image: image.png]

The map appears for region, and....the publish button is grayed out.
[image: image.png]
Also I'm unable to save it as a draft even though the button's appearance
hasn't changed. Just nothing happens when I click it.

Your help would be greatly appreciated--Thank you for responding to my
post!! :-)

**Reply by joshreisner** [reply]
hey @AZ-AA-Ron i'm not seeing any images - did you email them in? i think github doesn't process email attachments, you have to come to the page (https://github.com/code4recovery/12-step-meeting-list/discussions/480) to upload them

**Reply by AZ-AA-Ron** [reply]
Here we go: 
Let's say I want to edit a draft for this meeting--this is how it first
appears (notice "publish" button is still blue):
![image](https://github.com/code4recovery/12-step-meeting-list/assets/137969050/b0b33877-e50e-445e-8b21-638519e854d0)

I want to edit the status of in-person and make it online so I change it:
![image](https://github.com/code4recovery/12-step-meeting-list/assets/137969050/341cae80-b37b-415a-ad2e-8a36beb574c6)
![image](https://github.com/code4recovery/12-step-meeting-list/assets/137969050/4e0fcacf-d425-4723-aa25-cd1e757a50cc)

The map appears for region, and....the publish button is grayed out.
![image](https://github.com/code4recovery/12-step-meeting-list/assets/137969050/7aaf6e59-79f8-4088-8b01-3c977b8252ed)

Also I'm unable to save it as a draft even though the button's appearance
hasn't changed. Just nothing happens when I click it.

**Reply by joshreisner** [reply]
great, thanks @AZ-AA-Ron ! so you're going to need to enter a location and address for the meeting - with online meeting typically this is location = "Online Meeting" and address = "Flagstaff, AZ" (or some other approximate location) -- did you try that yet?

**Reply by AZ-AA-Ron** [reply]
I just did this--it seems to work??!

Thank you!!

**Reply by joshreisner** [reply]
great! it's really just the address that's required - we call that out the info panel below "can i attend this meeting in person?" - but having that location name in there as well is nice

**Comment by AZ-AA-Ron** [comment]
Oh ok. Yeah I pasted them in the body of the email and just replied to the
email.

**Comment by AZ-AA-Ron** [comment]
Thanks again, Josh. This was huge for this newb webservant. Continuity to
add and update meetings.

Thanks for being of service!!
